osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/DeviceChannelsHelper.java

```diff
@@ -110,28 +113,23 @@ protected void resetMBusClientAttributeValues(
       final String executorName)
       throws ProtocolAdapterException {
 
-    final DlmsObject mbusClientSetupObject =
-        this.dlmsObjectConfigService.getDlmsObject(device, DlmsObjectType.MBUS_CLIENT_SETUP);
-    final ObisCode obiscode = mbusClientSetupObject.getObisCodeWithChannel(channel);
-    final int classId = mbusClientSetupObject.getClassId();
-
     final DataObjectAttrExecutors dataObjectExecutors =
         new DataObjectAttrExecutors(executorName)
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.IDENTIFICATION_NUMBER, UINT_32_ZERO, obiscode, classId))
+                    MbusClientAttribute.IDENTIFICATION_NUMBER, UINT_32_ZERO, device, channel))
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.MANUFACTURER_ID, UINT_16_ZERO, obiscode, classId))
+                    MbusClientAttribute.MANUFACTURER_ID, UINT_16_ZERO, device, channel))
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.VERSION, UINT_8_ZERO, obiscode, classId))
+                    MbusClientAttribute.VERSION, UINT_8_ZERO, device, channel))
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.DEVICE_TYPE, UINT_8_ZERO, obiscode, classId))
+                    MbusClientAttribute.DEVICE_TYPE, UINT_8_ZERO, device, channel))
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.PRIMARY_ADDRESS, UINT_8_ZERO, obiscode, classId));
+                    MbusClientAttribute.PRIMARY_ADDRESS, UINT_8_ZERO, device, channel));
```

Comment:
The `getMbusAttributeExecutor` function will get the M-Bus client setup object from the object config every call again. Wouldn't it be better to get the object once and pass it to the function?

Reply:
CosemObjectAccessor for the MBUS_CLIENT_SETUP is now passed to the method so objectconfig is only read onxe